packages/@aws-cdk/aws-glue/lib/connection.ts

```diff
@@ -0,0 +1,193 @@
+
+  /**
+   * Designates a connection to a MongoDB document database.
+   */
+  public static readonly MONGODB = new ConnectionType('MONGODB');
+
+  /**
+   * Designates a network connection to a data source within an Amazon Virtual Private Cloud environment (Amazon VPC).
+   */
+  public static readonly NETWORK = new ConnectionType('NETWORK');
+
+  /**
+   * The name of this ConnectionType, as expected by Connection resource.
+   */
+  public readonly name: string;
+
+  constructor(name: string) {
+    this.name = name;
+    ConnectionType.ALL.push(this);
+  }
+
+  /**
+   * The connection type name as expected by Connection resource.
+   */
+  public toString(): string {
+    return this.name;
+  }
+}
+
+/**
+ * Interface representing a created or an imported {@link Connection}
+ */
+export interface IConnection extends cdk.IResource {
+  /**
+   * The name of the connection
+   * @attribute
+   */
+  readonly connectionName: string;
+
+  /**
+   * The ARN of the connection
+   * @attribute
+   */
+  readonly connectionArn: string;
+}
+
+/**
+ * Attributes for importing {@link Connection}
+ */
+export interface ConnectionAttributes {
+  /**
+   * The name of the connection
+   */
+  readonly connectionName: string;
+}
+
+/**
+ * Construction properties for {@link Connection}
+ */
+export interface ConnectionProps {
+  /**
+   * The name of the connection
+   * @default cloudformation generated name
+   */
+  readonly connectionName?: string;
+
+  /**
+   * The description of the connection.
+   * @default no description
+   */
+  readonly description?: string;
+
+  /**
+   * The type of the connection
+   */
+  readonly connectionType: ConnectionType;
+
+  /**
+   *  Key-Value pairs that define parameters for the connection.
+   *  @default empty properties
+   */
+  readonly connectionProperties?: { [key: string]: string };
+
+  /**
+   * A list of criteria that can be used in selecting this connection.
+   * @default no match criteria
+   */
+  readonly matchCriteria?: string[];
+
+  /**
+   * The list of security groups needed to successfully make this connection e.g. to successfully connect to VPC.
+   * @default no security group
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * The VPC subnet to connect to resources within a VPC. See more at https://docs.aws.amazon.com/glue/latest/dg/start-connecting.html.
+   * @default no subnet
+   */
+  readonly subnet?: ec2.ISubnet;
+}
+
+/**
+ * An AWS Glue connection to a data source.
+ */
+export class Connection extends cdk.Resource implements IConnection {
+
+  /**
+   * Creates a Connection construct that represents an external connection.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param attrs Import attributes
+   */
+  public static fromConnectionAttributes(scope: constructs.Construct, id: string, attrs: ConnectionAttributes): IConnection {
```

Comment:
split to `fromConnectionName` and `fromConnectionArn`. We usually reserve `fromXXAttributes` to allow importing additional runtime information, i.e those used beyond the ID of the resource.

Reply:
will do that